src/DynamoCore/Engine/EngineController.cs

```diff
@@ -69,7 +70,7 @@ private void OnTraceReconciliationComplete(TraceReconciliationEventArgs e)
         /// </summary>
         public static CompilationServices CompilationServices;
 
-        internal ProfilingSession profilingSession;
+        internal ProfilingSession profilingSession = null;
```

Comment:
Does the `ProfilingSession` need to be owned both by `AstBuilder` and `EngineController`?

Reply:
I have moved the ownership of the profiling data into the ASTBuilder only. Latest changes have just been pushed.